### DISCUSSION TITLE
capturing events in an AsyncSite

### DISCUSSION THREAD

**Discussion by a-ghorbani** [h2oai/wave#764] [Q&A]
Is it possible to handle an event (for example through `command`) from a card within a page that is created by AsyncSite()?

Further explanation: 
In the example given here: https://wave.h2o.ai/docs/examples/site-async/ l'd like to add a command to `ui.wide_gauge_stat_card` in this example. 
My question is how can I capture the event triggered by the command?
![Screenshot 2021-04-27 at 11 55 26](https://user-images.githubusercontent.com/11278140/116223539-eae40880-a74f-11eb-8eb0-54e713af90ec.png)

**Comment by lo5** [comment]
@a-ghorbani If your app is hosted at `/foo` and the card is hosted at `/bar`, there is currently no way for the app to handle events from the card. 

That particular example was intended to be a demonstration of how to make an app display non-interactive pages at URLs other than its own.

**Reply by lo5** [reply]
Thinking further, there are 2 solutions here (if I understand your problem correctly)
1. Make the app (hosted at `/foo`) display the cards at `/foo#bar` instead `/bar`. That way, the app can handle all events under `/foo*`.
2. **New feature:** make the `ui.command(...)` (in your case, "Learn more") support paths (`/foo/stats/42`) so that clicking on the menu item navigates your browser to `/foo/stats/42`, where your app can display more details about that card).
  a. **New feature:** make the card itself clickable, so that you can simplify the UI and avoid having to add a menu item.

**Reply by a-ghorbani** [reply]
Thanks @lo5 

2.a would be awesome to have.
